src/Sylius/Component/Order/Model/OrderItem.php

```diff
@@ -131,175 +130,214 @@ public function setUnitPrice($unitPrice)
         if (!is_int($unitPrice)) {
             throw new \InvalidArgumentException('Unit price must be an integer.');
         }
+
         $this->unitPrice = $unitPrice;
+        $this->calculateTotal();
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getAdjustments($type = null)
+    public function getTotal()
     {
-        if (null === $type) {
-            return $this->adjustments;
-        }
-
-        return $this->adjustments->filter(function (AdjustmentInterface $adjustment) use ($type) {
-            return $type === $adjustment->getType();
-        });
+        return $this->total;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function addAdjustment(AdjustmentInterface $adjustment)
+    public function calculateTotal()
     {
-        if (!$this->hasAdjustment($adjustment)) {
-            $adjustment->setAdjustable($this);
-            $this->adjustments->add($adjustment);
+        $this->calculateAdjustmentsTotal();
+
+        $this->total = $this->adjustmentsTotal;
+        $this->total += $this->getItemUnitsTotal();
+
+        if ($this->total < 0) {
+            $this->total = 0;
         }
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function removeAdjustment(AdjustmentInterface $adjustment)
+    protected function getItemUnitsTotal()
```

Comment:
No policy :smile:  I think we will introduce the `itemUnitsTotal` filed on the class and the getter will be public for sure :wink: naming and behavior of recalculation methods will be change greatly (ref. **Performance** point :)).